### PR TITLE
BUGZ-85 - audio pipeline interpolation on ring buffer starve

### PR DESCRIPTION
**Pull request by roxanneskelly**
Kick the PLC so that it generates a more pleasing 'fade' frame
when ring buffer starves.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19244/HighFidelity-Beta-PR15518-93a4654.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19244/HighFidelity-Quest-Beta-PR15518-93a4654.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19244/HighFidelity-Beta-PR15518-93a4654.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19244/HighFidelity-Beta-Interface-PR15518-93a4654.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19244/HighFidelity-Beta-PR15518-93a4654.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19244/HighFidelity-Beta-Interface-PR15518-93a4654.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19286/HighFidelity-Beta-PR15518-8a4eb7a.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19286/HighFidelity-Quest-Beta-PR15518-8a4eb7a.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19286/HighFidelity-Beta-PR15518-8a4eb7a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19286/HighFidelity-Beta-Interface-PR15518-8a4eb7a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19286/HighFidelity-Beta-PR15518-8a4eb7a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19286/HighFidelity-Beta-Interface-PR15518-8a4eb7a.exe

File: libraries/audio/src/InboundAudioStream.cpp [hunk 1]
```diff
@@ -338,10 +359,23 @@ int InboundAudioStream::popSamples(int maxSamples, bool allOrNothing) {
...
+            } else {
+                // No samples available means a packet is currently being
+                // processed, so we don't generate lost audio data, and instead
+                // just wait for the packet to come in.  This prevents locking
+                // the real-time audio thread at the cost of a potential (but rare)
+                // 'click'
+                _lastPopSucceeded = false;
+            }
         }
     }
```
Comment (Atlante45) [inline libraries/audio/src/InboundAudioStream.cpp]: I'm really confused by that extra bracket.

**Pull request closed without merging**